util-core/src/main/scala/com/twitter/util/Batcher.scala

```diff
@@ -0,0 +1,20 @@
+package com.twitter.util
+
+/** Provides a clean, lightweight interface for working with batched Futures */
+private[util] class Batcher[In, Out](
+  executor: BatchExecutor[In, Out]
+)(
+  implicit timer: Timer
+) extends Function1[In, Future[Out]] { batcher =>
+  /** Enqueues requests for a batched Future */
```

Comment:
feel free to ignore:
Maybe also note that if apply makes it hit the batch size, it will automatically process the batch.

s/Future/[[com.twitter.util.Future]]